{
  "cveId": "CVE-2025-3248",
  "title": "Langflow Unauth RCE",
  "description": "Langflow versions prior to 1.3.0 are susceptible to code injection in \nthe /api/v1/validate/code endpoint. A remote and unauthenticated attacker can send crafted HTTP requests to execute arbitrary\ncode.",
  "publishedDate": "2025-04-07T14:22:38.980000+00:00",
  "lastModifiedDate": "2025-05-05T22:20:22.724000+00:00",
  "severity": "CRITICAL",
  "cvssMetrics": [
    {
      "version": "3.1",
      "vectorString": "CVSS:3.1/AV:N/AC:L/PR:N/UI:N/S:U/C:H/I:H/A:H",
      "baseScore": 9.8,
      "baseSeverity": "CRITICAL"
    }
  ],
  "epss": {
    "score": 0.9239,
    "percentile": 99.714
  },
  "riskScore": 67,
  "exploitationStatus": "UNKNOWN",
  "affectedSystems": {
    "cpeMatches": [],
    "vendors": ["langflow-ai"],
    "products": ["langflow"]
  },
  "references": [
    {
      "url": "https://github.com/langflow-ai/langflow/pull/6911",
      "source": "",
      "tags": ["patch"]
    },
    {
      "url": "https://github.com/langflow-ai/langflow/releases/tag/1.3.0",
      "source": "",
      "tags": ["release-notes"]
    },
    {
      "url": "https://www.horizon3.ai/attack-research/disclosures/unsafe-at-any-speed-abusing-python-exec-for-unauth-rce-in-langflow-ai/",
      "source": "",
      "tags": ["exploit"]
    }
  ],
  "attackTechniques": [],
  "metadata": {
    "tags": ["CWE-306"],
    "requiresUserInteraction": null,
    "requiresPrivileges": null,
    "attackVector": null,
    "sources": [
      {
        "name": "CVEList",
        "url": "https://github.com/CVEProject/cvelistV5/blob/main/cves/2025/3xxx/CVE-2025-3248.json",
        "lastModified": "2025-05-05T22:20:22.724000+00:00"
      }
    ]
  }
}
